extensions/amp-selector/0.1/test/test-amp-selector.js

```diff
@@ -132,6 +136,25 @@ describes.realWin('amp-selector', {
       expect(initSpy).to.be.calledOnce;
     });
 
+    it('should retain existing roles', function* () {
+      const ampSelector = getSelector({
+        attributes: {
+          role: 'tablist',
+        },
+        optionAttributes: {
+          role: 'tab',
+        },
+        config: {
+          count: 4,
+          selectedCount: 2,
+        },
+      });
+      const impl = ampSelector.implementation_;
+      yield ampSelector.build();
```

Comment:
I think `build` is sync.

Reply:
Looks like the `buildCallback` portion can be async if you have `data-block-on-consent`.


https://github.com/ampproject/amphtml/blob/069656c9b29cd608a4223fd8c826c2fad27b233c/src/custom-element.js#L467-L502

There is some async work in the promise chain too. Might be a good idea in general to use the promise. For the most part, just following the pattern of the rest of the file though, not sure that it matters here.